hutch_python/log_setup.py

```diff
@@ -269,16 +466,17 @@ def whitelist_all_level(self) -> str:
         return logging.getLevelName(self._whitelist_all_levelno)
 
     @whitelist_all_level.setter
-    def whitelist_all_level(self, value):
+    def whitelist_all_level(self, value: Union[int, str]):
         self._whitelist_all_levelno = validate_log_level(value)
 
     @property
-    def object_names(self):
+    def object_names(self) -> set[str]:
         """The names of all contained objects."""
         return set(obj.name for obj in self.objects)
 
-    def filter(self, record):
-        if not hasattr(record, 'ophyd_object_name'):
+    def filter(self, record: logging.LogRecord) -> bool:
+        name: Optional[str] = getattr(record, "ophyd_object_name", None)
```

Comment:
For some reason I never considered adding type annotations to variables inside function scopes. I might start doing this.